tests/perf/conftest.py

```diff
@@ -0,0 +1,196 @@
+        #
+        # Instead, they run either with the BENCH_DIR specified explicitly
+        # both as a target and a 'confcutdir' or when the BENCH_DIR itself
+        # is a 'rootdir' (CWD). In any case this conftest is not processed
+        # at all, and this hook doesn't run when running tests from the
+        # BENCH_DIR.
+        return True
+
+
+def pytest_itemcollected(item):
+    """This is only called for local tests (under tests/perf_runner/)."""
+    if path_in_dir(item.fspath):
+        item.add_marker('_perf_test')
+
+
+def pytest_collection_modifyitems(session, config, items):
+    """This is called after collection (of all tests) has been performed."""
+    run_perf = config.getoption('run_perf')
+    if run_perf in ('yes', 'check'):
+        return
+    perf_only = (run_perf == 'only')
+
+    items[:] = [item
+                for item in items
+                if bool(item.get_marker('_perf_test')) == perf_only]
+
+
+def pytest_terminal_summary(terminalreporter):
+    """Add additional section in terminal summary reporting."""
+    config = terminalreporter.config
+
+    if config.getoption('run_perf') in ('yes', 'only'):
+        passed = terminalreporter.stats.get('passed', [])
+        for rep in passed:
+            if '_perf_test' not in rep.keywords:
+                continue
+
+            out = '\n'.join(s for _, s in rep.get_sections('Captured stdout'))
+            if out:
+                fspath, lineno, name = rep.location
+                terminalreporter.write_sep("-", 'Report from {0}'.format(name))
+                terminalreporter.write(out)
+
+    handle_perf_graph(config, terminalreporter)
+
+
+def handle_perf_graph(config, terminalreporter):
+    output_file = config.getoption('perf_graph_name')
+    if not output_file:
+        return
+
+    bench_storage = config.getoption('--benchmark-storage')
+    output_file = os.path.join(bench_storage, output_file)
+
+    trial_names, benchmarks = load_benchmarks(bench_storage,
+                                              modes=sorted(mode_args_map))
+    if not trial_names:
+        terminalreporter.write_line(
+            'perf-graph: No benchmarks found in {0}'.format(bench_storage),
+            yellow=True, bold=True)
+        return
+
+    expr = config.getoption('perf_expr_primary')
+    expr2 = config.getoption('perf_expr_secondary')
+
+    history = eval_benchmark_expr(expr, benchmarks)
+    history2 = eval_benchmark_expr(expr2, benchmarks)
+
+    plot = make_plot(
+        trial_names=trial_names,
+        history=history,
+        history2=history2,
+        expr=expr,
+        expr2=expr2,
+    )
+    plot.render_to_file(output_file)
+
+    terminalreporter.write_line(
+        'perf-graph: Saved graph into {0}'.format(output_file),
+        green=True, bold=True)
+
+
+@gen_dict
+def eval_benchmark_expr(expr, benchenvs, **kwargs):
+    for mode, envlist in benchenvs.items():
+        yield mode, [BenchmarkEnv(benchenv, **kwargs).evaluate(expr)
```

Comment:
Indeed it can, and this is intentional. Pygal takes care of `None` values properly treating that as missing measurements data.